docs/data/date-pickers/validation/DateValidationDisablePast.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import { DateRange, DateRangePicker } from '@mui/x-date-pickers-pro/DateRangePicker';
+
+const today = dayjs();
+const yesterday = dayjs().subtract(1, 'day');
+
+const GridItem = ({
+  label,
+  children,
+  spacing = 1,
+}: {
+  label: string;
+  children: React.ReactNode;
+  spacing?: number;
+}) => {
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
+      <Typography variant="body2" sx={{ mb: spacing }}>
+        {label}
+      </Typography>
+      {children}
+    </Box>
+  );
+};
+
+export default function DateValidationDisablePast() {
+  const [datePickerValue, setDatePickerValue] = React.useState<Dayjs | null>(
+    yesterday,
+  );
+  const [dateTimePickerValue, setDateTimePickerValue] = React.useState<Dayjs | null>(
+    yesterday,
+  );
+  const [dateRangePickerValue, setDateRangePickerValue] = React.useState<
+    DateRange<Dayjs>
+  >([yesterday, today]);
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={4}>
```

Comment:
![image](https://user-images.githubusercontent.com/3309670/190981263-3baf73c4-c3a0-47c1-a164-026c6df24884.png)

Here is the result with this diff
I'll push it for the `DateValidationDisablePast` demo.

We clearly gain a lot of space
But on the other hand I find it harder to distinguish the various pickers (see that one line 1 it's 2 pickers with one input each but on line 2 it's 1 picker with two inputs).

Reply:
I'm in favour of the `disablePast` variant and am for replacing layout to such for all the demos containing 2 or 3 inputs. 🤔 
What do you think @joserodolfofreitas @alexfauquette @samuelsycamore ?

Reply:
I like the look of the `disableFuture` examples better. I would much prefer to see each example on its own line. Makes it easier to quickly skim and understand what you're looking at.

It's also important to consider the difference in UX between light and dark mode in the docs - what looks like decent spacing in light mode can appear too cluttered in dark mode, so I think it's important to err on the side of too much space rather than not enough.